tests/integration/profile_test.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import os
+
+
+from contextlib import contextmanager
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+
+import pytest
+
+from .testlib import cmd as libcmd
+
+
+DUMMY0_IFNAME = 'dummy0'
+
+NMCLI_CON_ADD_CMD = [
+    'nmcli',
+    'con',
+    'add',
+    'type',
+    'dummy',
+    'con-name',
+    'testProfile',
+    'connection.autoconnect',
+    'no',
+    'ifname',
+    DUMMY0_IFNAME,
+]
+
+PROFILE_DIRECTORY = '/etc/NetworkManager/system-connections/'
+
+
+def test_delete_inactive_profiles(dummy_prepare):
```

Comment:
I think this test will cover only one part of the production code. The other part is for editing an existing iface which has a non active profile.

This test is about a profile with a non existent iface.
When you add the other test, you’ll need to specify this in the name of the current test name.

Reply:
I've create a new test and renamed the existing one. Thanks!